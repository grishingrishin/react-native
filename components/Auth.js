import React, { Component } from 'react';
import styled from 'styled-components';

const Wrapper = styled.View`
  flex: 1;
  background-color: #693fa9;
`;

const Container = styled.ImageBackground`
  flex: 1;
  padding-top: 32px;
  padding-bottom: 32px;
  justify-content: space-between;
  resize-mode: cover;
`;

const Header = styled.View`
  position: relative;
  top: 15%;
`;

const Title = styled.Text`
  text-align: center;
  font-size: 42px;
  color: #fff;
`;

const WrapperInput = styled.View`
  padding-right: 16px;
  padding-left: 16px;
`;

const TextInput = styled.TextInput`
  padding-top: 16px;
  padding-right: 14px;
  padding-bottom: 16px;
  padding-left: 14px;
  border-bottom-width: 1px;
  border-bottom-color: #d3c8e6;
`;

const Forgot = styled.TouchableOpacity`
  align-items: flex-end;
`;

const ForgotText = styled.Text`
  text-transform: uppercase;
  padding-top: 14px;
  padding-bottom: 14px;
  font-size: 12px;
  color: #fff;
`;

const Action = styled.View``;

const Submit = styled.TouchableOpacity`
  align-items: center;
  height: 54px;
  background-color: #eb2054;
`;

const SubmitText = styled.Text`
  text-transform: uppercase;
  font-size: 18px;
  line-height: 54px;
  color: #fff;
`;

class Auth extends Component {
  constructor() {
    super();

    this.state = {
      data: {
        login: '',
        password: ''
      }
    };
  }

  changeLoginHandler = text => {
    return this.setState({
      data: {
        ...this.state.data,
        login: text
      }
    });
  }

  changePasswordHandler = text => {
    return this.setState({
      data: {
        ...this.state.data,
        password: text
      }
    });
  }

  goToForgotPassword = () => {}

  loginHandler = () => {}

  render() {
    return (
      <Wrapper>
        <Container source={require('../assets/images/auth-bg.jpg')}>
          <Header>
            <Title accessibilityRole='header'>Welcome!</Title>
          </Header>
          <WrapperInput>
            <TextInput
              onChangeText={text => this.changeLoginHandler(text)}
              value={this.state.data.login}
              placeholder='Login'
            />
            <TextInput 
              onChangeText={text => this.changePasswordHandler(text)}
              value={this.state.data.password}
              secureTextEntry={true}
              placeholder='Password'
            />
            <Forgot onPress={this.goToForgotPassword}>
              <ForgotText>Forgot Password?</ForgotText>
            </Forgot>
          </WrapperInput>
          <Action>
            <Submit activeOpacity={0.9} onPress={this.loginHandler}>
              <SubmitText>Login</SubmitText>
            </Submit>
          </Action>
        </Container>
      </Wrapper>
    );
  }
}

export default Auth;