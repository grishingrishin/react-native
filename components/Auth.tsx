import React, {Component} from 'react';
import {TouchableOpacity} from 'react-native';
import {Link} from 'react-router-native';
import styled from 'styled-components/native';
import FadeInLoader from './FadeInLoader';
import FloatingLabelInput from './FloatingLabelInput';
import Question from './Question';

const Container = styled.ImageBackground`
  flex: 1;
  padding: 32px 0;
  justify-content: space-between;
  resize-mode: cover;
`;

const Header = styled.View`
  position: relative;
  top: 15%;
`;

const Title = styled.Text`
  text-align: center;
  font-size: 42px;
  font-family: Roboto-Thin;
  color: #fff;
`;

const Box = styled.View`
  padding: 0 16px;
`;

const Group = styled.View`
  margin-bottom: 14px;
`;

const Forgot = styled.View`
  align-items: flex-end;
`;

const ForgotText = styled.Text`
  text-transform: uppercase;
  padding: 14px 0;
  font-size: 10px;
  color: #fff;
`;

const Actions = styled.View``;

const Submit = styled.TouchableOpacity`
  align-items: center;
  height: 54px;
  background-color: #eb2054;
`;

const SubmitText = styled.Text`
  text-transform: uppercase;
  font-size: 18px;
  line-height: 54px;
  font-family: Roboto-Regular;
  color: #fff;
`;

interface AState {
  login: string;
  password: string;
}

class Auth extends Component<{}, AState> {
  constructor(props: object) {
    super(props);

    this.state = {
      login: '',
      password: '',
    };
  }

  private handleLogin = (login: string): void => {
    this.setState({
      login,
    });
  };

  private handlePassword = (password: string): void => {
    this.setState({
      password,
    });
  };

  public render() {
    return (
      <Container source={require('../assets/auth-bg.jpg')}>
        <Header>
          <Title>Welcome!</Title>
        </Header>
        <FadeInLoader>
          <Box>
            <Group>
              <FloatingLabelInput
                value={this.state.login}
                placeholder="Login"
                changeHandle={this.handleLogin}
              />
            </Group>
            <Group>
              <FloatingLabelInput
                value={this.state.password}
                placeholder="Password"
                secureTextEntry={true}
                changeHandle={this.handlePassword}
              />
              <Forgot>
                <Link to="/forgot" component={TouchableOpacity}>
                  <ForgotText>Forgot Password?</ForgotText>
                </Link>
              </Forgot>
            </Group>
          </Box>
        </FadeInLoader>
        <Actions>
          <Submit activeOpacity={0.9}>
            <SubmitText>Sing in</SubmitText>
          </Submit>
          <Question
            path="/registry"
            route="Register now!"
            question="Still donâ€™t have an account?"
          />
        </Actions>
      </Container>
    );
  }
}

export default Auth;
