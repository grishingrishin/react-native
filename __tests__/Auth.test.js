import React from 'react';
import {shallow} from 'enzyme';
import {create} from 'react-test-renderer';
import Auth from '../components/Auth';

// Disable warning: `useNativeDriver` is not supported because the native animated module is missing. Falling back to JS-based animation.
// To resolve this, add `RCTAnimation` module to this app, or remove `useNativeDriver.
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
// Fix referenceError: You are trying to `import` a file after the Jest environment has been torn down.
jest.useFakeTimers();

describe('<Auth />', () => {
  describe('Jest tests', () => {
    it('renders correctly', () => {
      const tree = create(<Auth />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Enzyme tests', () => {
    it('allows us to set some state', () => {
      const wrapper = shallow(<Auth />);

      wrapper.setState({
        login: 'test@test',
        password: 'testtesttest',
      });

      expect(wrapper.state().login).toEqual('test@test');
      expect(wrapper.state().password).toEqual('testtesttest');
    });

    it('allows us to simulate change event', () => {
      const wrapper = shallow(<Auth />);

      expect(wrapper.find('FloatingLabelInput')).toHaveLength(2);

      wrapper
        .find('FloatingLabelInput')
        .first()
        .props()
        .changeHandle('test@test');
      expect(wrapper.state().login).toEqual('test@test');

      wrapper
        .find('FloatingLabelInput')
        .last()
        .props()
        .changeHandle('testtesttest');
      expect(wrapper.state().password).toEqual('testtesttest');

      expect(wrapper).toMatchSnapshot();
    });
  });
});
