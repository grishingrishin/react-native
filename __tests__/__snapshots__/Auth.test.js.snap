// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> Enzyme tests allows us to simulate change event 1`] = `
<Auth__Container
  source={1}
>
  <Auth__Header>
    <Auth__Title>
      Welcome!
    </Auth__Title>
  </Auth__Header>
  <FadeInLoader>
    <Auth__Box>
      <Auth__Group>
        <FloatingLabelInput
          changeHandle={[Function]}
          placeholder="Login"
          secureTextEntry={false}
          value="test@test"
        />
      </Auth__Group>
      <Auth__Group>
        <FloatingLabelInput
          changeHandle={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          value="testtesttest"
        />
        <Auth__Forgot>
          <Link
            component={[Function]}
            replace={false}
            to="/forgot"
          >
            <Auth__ForgotText>
              Forgot Password?
            </Auth__ForgotText>
          </Link>
        </Auth__Forgot>
      </Auth__Group>
    </Auth__Box>
  </FadeInLoader>
  <Auth__Actions>
    <Auth__Submit
      activeOpacity={0.9}
    >
      <Auth__SubmitText>
        Sing in
      </Auth__SubmitText>
    </Auth__Submit>
    <_default
      path="/registry"
      question="Still don’t have an account?"
      route="Register now!"
    />
  </Auth__Actions>
</Auth__Container>
`;

exports[`<Auth /> Jest tests renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "paddingBottom": 32,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 32,
        "resizeMode": "cover",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "position": "relative",
          "top": "15%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Roboto-Thin",
            "fontSize": 42,
            "textAlign": "center",
          },
        ]
      }
    >
      Welcome!
    </Text>
  </View>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginBottom": 14,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "left": 14,
                "opacity": 1,
                "position": "absolute",
                "top": 18,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Roboto-Light",
                  },
                ]
              }
            >
              Login
            </Text>
          </Text>
          <TextInput
            allowFontScaling={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderBottomColor": "rgba(211,200,230,0.5)",
                  "borderBottomWidth": 1,
                  "color": "#fff",
                  "fontFamily": "Roboto-Light",
                  "fontSize": 16,
                  "paddingBottom": 18,
                  "paddingLeft": 14,
                  "paddingRight": 14,
                  "paddingTop": 18,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 14,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "left": 14,
                "opacity": 1,
                "position": "absolute",
                "top": 18,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Roboto-Light",
                  },
                ]
              }
            >
              Password
            </Text>
          </Text>
          <TextInput
            allowFontScaling={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "borderBottomColor": "rgba(211,200,230,0.5)",
                  "borderBottomWidth": 1,
                  "color": "#fff",
                  "fontFamily": "Roboto-Light",
                  "fontSize": 16,
                  "paddingBottom": 18,
                  "paddingLeft": 14,
                  "paddingRight": 14,
                  "paddingTop": 18,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                    "paddingBottom": 14,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 14,
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              Forgot Password?
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#eb2054",
          "height": 54,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
              "lineHeight": 54,
              "textTransform": "uppercase",
            },
          ]
        }
      >
        Sing in
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 18,
            "textAlign": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto-Thin",
              "fontSize": 12,
            },
          ]
        }
      >
        Still don’t have an account?
         
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Roboto-Regular",
                "fontSize": 12,
                "textDecorationColor": "black",
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
              },
            ]
          }
        >
          Register now!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
